src/perform_step/rosenbrock_perform_step.jl

```diff
@@ -1061,11 +1061,15 @@ end
         integrator.k[2] = h31 * k1 + h32 * k2 + h33 * k3 + h34 * k4 + h35 * k5
         integrator.k[3] = h2_21 * k1 + h2_22 * k2 + h2_23 * k3 + h2_24 * k4 + h2_25 * k5
         if integrator.opts.adaptive
-            calculate_interpoldiff!(
-                k1, k2, uprev, du, u, integrator.k[1], integrator.k[2], integrator.k[3])
-            atmp = calculate_residuals(k2, uprev, k1, integrator.opts.abstol,
+            if  isa(linsolve_tmp,AbstractFloat)
+                u_int, u_diff = calculate_interpoldiff(uprev, du, u, integrator.k[1], integrator.k[2], integrator.k[3])
+            else
+                u_int = linsolve_tmp; u_diff = copy(linsolve_tmp)
```

Comment:
This should get a spot in the cache, or reuse another temporary.

Reply:
I'm not sure if the "else branch" will ever go through. This version of perform_step seems to be for scalar problems only, the line below also allocates:
 
atmp = calculate_residuals....